src/file.cr

```diff
@@ -866,6 +866,12 @@ class File < IO::FileDescriptor
     end
   end
 
+  # Rename the current `File`
+  def rename(new_filename : Path | String) : Nil
+    File.rename(@path, new_filename)
+    @path = new_filename.to_s
```

Comment:
Do the both cases the same?
`new_filename` could be just a file name (as the test covers)) but could be a (full) path...
BTW is `@path` a (stringified) path or a file name?

Reply:
`@path` is whatever you pass to the constructor. It can be a single file name or consist of multiple path components. It makes no difference which.